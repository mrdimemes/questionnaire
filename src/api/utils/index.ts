export { default as getFetchError } from "./getFetchError";