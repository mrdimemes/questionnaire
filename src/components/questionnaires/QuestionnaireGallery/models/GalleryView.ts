export enum GalleryView {
  Rows = "ROWS",
  Plates = "PLATES",
};