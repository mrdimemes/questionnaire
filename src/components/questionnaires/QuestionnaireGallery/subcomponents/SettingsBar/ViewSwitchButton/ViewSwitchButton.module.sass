@use "src/sass/foundation/mixins"
@use "src/sass/foundation/colors"

.ViewSwitchButton
  @include mixins.square(25px)
  @include mixins.noborder
  padding: 0
  &.plates
    @include mixins.mask-image("/assets/images/icons/plates.svg")
  &.rows
    @include mixins.mask-image("/assets/images/icons/rows.svg")

.theme_dark
  background-color: colors.$dark-theme-4

.theme_light
  background-color: colors.$gray-1