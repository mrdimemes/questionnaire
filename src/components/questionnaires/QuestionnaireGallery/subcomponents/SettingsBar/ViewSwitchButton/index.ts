export { default as ViewSwitchButton } from "./ViewSwitchButton";