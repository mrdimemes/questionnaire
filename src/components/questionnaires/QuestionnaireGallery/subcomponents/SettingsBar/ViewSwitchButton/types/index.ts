export type { ViewSwitchProps } from "./ViewSwitchProps";