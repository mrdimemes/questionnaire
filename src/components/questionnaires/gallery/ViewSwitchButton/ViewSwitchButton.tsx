import classNames from "classnames";
import { useAppSelector } from "src/redux/hooks";
import { themeSelector } from "src/redux/selectors";
import { getThemeStyle } from "src/redux/slices/themeSlice";
import { Button } from "src/components/UI/Button";
import { GalleryViews } from "../QuestionnaireGallery/QuestionnaireGallery";
import Plates from "src/assets/images/icons/plates.svg";
import Rows from "src/assets/images/icons/rows.svg";
import styles from "./ViewSwitchButton.module.sass";

type ViewSwitchProps = {
  setCurrentView: (newView: GalleryViews) => void,
  currentView: GalleryViews
};

const ViewSwitchButton = (
  { setCurrentView, currentView }: ViewSwitchProps
) => {
  const currentTheme = useAppSelector(themeSelector);
  const toggleView = () => {
    switch (currentView) {
      case GalleryViews.Plates:
        setCurrentView(GalleryViews.Rows);
        break;
      case GalleryViews.Rows:
        setCurrentView(GalleryViews.Plates)
        break;
    }
  };

  return (
    <Button
      onClick={toggleView}
      className={classNames(
        styles.body,
        getThemeStyle(styles, currentTheme)
      )}
    >
      <img
        src={currentView === GalleryViews.Plates ? Rows : Plates}
        alt="view mode"
      />
    </Button>
  )
}

export default ViewSwitchButton