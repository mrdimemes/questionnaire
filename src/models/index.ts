export { default as FetchStatus } from "./FetchStatus";
export { default as Tag } from "./Tag";
export { default as User } from "./User";
export { default as SortOption } from "./SortOption";
export { default as ValidationWarning } from "./ValidationWarning";

export { AuthOption } from "./auth";
export type { LoginResponse, RefreshResponse } from "./auth";

export {
  QuestionType,
  FieldAnswerDTO,
  QuestionAnswerDTO,
  QuestionnaireAnswerDTO,
  Field,
  Question,
  Questionnaire,
} from "./questionnaire";
export type {
  QuestionnaireCard,
  QuestionnaireCardsBunch,
} from "./questionnaire";