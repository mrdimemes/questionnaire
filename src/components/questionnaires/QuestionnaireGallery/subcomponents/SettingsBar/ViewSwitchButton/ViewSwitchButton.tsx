import classNames from "classnames";

import { Button } from "src/components";
import { withTheme } from "src/HOCs";

import { GalleryView } from "../../../models";

import styles from "./ViewSwitchButton.module.sass";

import type { ViewSwitchProps } from "./types";


const ViewSwitchButton = ({
  setCurrentView,
  currentView,
  className,
}: ViewSwitchProps) => {

  const toggleView = () => {
    setCurrentView(
      currentView === GalleryView.Plates ?
        GalleryView.Rows :
        GalleryView.Plates,
    );
  };

  return (
    <Button
      onClick={toggleView}
      className={classNames(
        styles.ViewSwitchButton,
        currentView === GalleryView.Plates ? styles.plates : styles.rows,
        className,
      )}
    />
  );
};

export default withTheme(ViewSwitchButton, styles);