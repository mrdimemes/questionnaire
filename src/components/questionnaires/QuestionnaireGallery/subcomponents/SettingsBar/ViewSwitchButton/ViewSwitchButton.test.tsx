import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderWithProviders } from "src/utils/test-utils";

import { GalleryView } from "../../../models";

import ViewSwitchButton from "./ViewSwitchButton";

let currentView: GalleryView;
const setCurrentView = (newView: GalleryView) => {
  currentView = newView;
};

beforeEach(() => setCurrentView(GalleryView.Rows));

describe("QuestionnaireGallery/ViewSwitchButton component", () => {
  test("ViewSwitchButton renders", () => {
    renderWithProviders(
      <ViewSwitchButton
        setCurrentView={setCurrentView}
        currentView={currentView}
      />,
    );
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("ViewSwitchButton switches the view", () => {
    renderWithProviders(
      <ViewSwitchButton
        setCurrentView={setCurrentView}
        currentView={currentView}
      />,
    );
    userEvent.click(screen.getByRole("button"));
    expect(currentView).toEqual(GalleryView.Plates);
  });

  test("ViewSwitchButton snapshot", () => {
    const { container } = renderWithProviders(
      <ViewSwitchButton
        setCurrentView={setCurrentView}
        currentView={currentView}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});