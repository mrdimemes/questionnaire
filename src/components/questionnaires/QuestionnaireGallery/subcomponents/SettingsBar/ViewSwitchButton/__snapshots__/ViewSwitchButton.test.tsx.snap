// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnaireGallery/ViewSwitchButton component ViewSwitchButton snapshot 1`] = `
<div>
  <button
    class="Button ViewSwitchButton rows theme_light theme_light"
  />
</div>
`;
